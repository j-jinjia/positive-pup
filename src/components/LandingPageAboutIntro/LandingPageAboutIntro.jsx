import "./LandingPageAboutIntro.scss";
import introImage from "../../assets/images/landing_page/landing_page_about_intro_img.png";
import Button from "../Button/Button";
import paw from "../../assets/images/landing_page/paw_print.svg";
import { Link } from "react-router-dom";

const LandingPageAboutIntro = () => {
  return (
      <div className="landing-page-about-intro">
          <img
            src={paw}
            className="landing-page-about-intro__paw"
            alt="paw print"
          />

          <h2 className="landing-page-about-intro__header">
            Why The Positive Pup?
          </h2>
          <p className="landing-page-about-intro__paragraph">
            {" "}
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mi
            malesuada tortor, tristique aenean ultricies. Integer aliquam mattis
            turpis justo, faucibus bibendum sagittis lacus ornare. Porttitor
            eget sed amet malesuada est dignissim magna. Amet vel tempor egestas
            at in mi, facilisi proin.
          </p>

            <Link to="/about" className="landing-page-about-intro__button">
              <Button isSecondary={true} buttonText="More About Me" />
            </Link>
            <img
              src={introImage}
              className="landing-page-about-intro__image"
              alt="Positive Pup owner, Melanie, with a dog"
            />
      </div>
  );
};

export default LandingPageAboutIntro;
