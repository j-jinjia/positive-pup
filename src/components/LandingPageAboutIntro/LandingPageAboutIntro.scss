@import "../../styles/variables";

@media screen and (min-width: $desktop-breakpoint) {
  .landing-page__about-intro {
    display: flex;
    flex-direction: column;

    &--header {
      display: flex;
      flex-direction: column;
      font-family: $font-header;
      font-size: 40px;
      font-weight: 500;
      padding: 211px 222px 0px 430px;
    }

    &--paragraph {
      display: flex;
      flex-direction: column;
      position: absolute;
      width: 524px;
      height: 180px;
      left: 430px;
      top: 305px;
      font-family: $font-general;
      font-size: 20px;
      line-height: 180%;
      color: $colour-font-paragraph;
    }

    &--greenbox {
      display: flex;
      flex-direction: column;
      background-color: rgba(55, 175, 162, 0.06);
      margin-left: 505px;
      width: 1126px;
      height: 734px;
      position: relative;
    }

    &--paw {
      display: flex;
      flex-direction: column;
      position: absolute;
      left: 430px;
      top: 145px;
      width: 60px;
      height: 60px;
    }

    &--image {
      display: flex;
      flex-direction: column;
      position: absolute;
      top: 78px;
      left: -177px;
    }

    &--button {
      display: flex;
      flex-direction: column;
      position: absolute;
      left: 430px;
      top: 529px;
      text-decoration: none;
    }
  }
}

@media (min-width: $tablet-breakpoint) and (max-width: $desktop-breakpoint) {
  .landing-page__about-intro {
    display: flex;
    flex-direction: column;
    .buttonAndImage-container {
      display: flex;
      justify-content: space-between;
      margin: 0 4rem;
    }

    &--header {
      display: flex;
      flex-direction: column;
      justify-content: center;
      text-align: center;
      align-items: center;
      font-family: $font-header;
      font-size: 1.75rem;
      font-weight: 600;
      line-height: 150%;
      padding-top: 71px;
    }

    &--paragraph {
      display: flex;
      flex-direction: column;
      margin-top: 22px;
      margin-bottom: 4rem;
      width: auto;
      font-family: $font-general;
      font-size: 18px;
      color: $colour-font-paragraph;
      line-height: 180%;
      padding: 1.5rem 4rem;
    }

    &--greenbox {
      display: flex;
      flex-direction: column;
      background-color: rgba(55, 175, 162, 0.06);
      width: 100%;
      height: auto;
      position: relative;
    }

    &--paw {
      display: none;
    }

    &--image {
      display: flex;
      flex-direction: column;
      width: 300px;
    }

    &--button {
      text-decoration: none;
    }
  }
}

@media (min-width: $mobile-breakpoint) and (max-width: $tablet-breakpoint) {
  .landing-page__about-intro {
    display: flex;
    flex-direction: column;
    .buttonAndImage-container {
      display: flex;
      flex-direction: column;
    }

    &--header {
      display: flex;
      flex-direction: column;
      justify-content: center;
      text-align: center;
      align-items: center;
      font-family: $font-header;
      font-size: 28px;
      font-weight: 600;
      line-height: 150%;
      padding-top: 71px;
    }

    &--paragraph {
      display: flex;
      flex-direction: column;
      margin-top: 22px;
      margin-left: 41px;
      width: auto;
      height: 288px;
      font-family: $font-general;
      font-size: 18px;
      color: $colour-font-paragraph;
      line-height: 180%;
    }

    &--greenbox {
      display: flex;
      flex-direction: column;
      background-color: rgba(55, 175, 162, 0.06);
      width: 100%;
      height: auto;
      position: relative;
    }

    &--paw {
      display: none;
    }

    &--image {
      display: flex;
      flex-direction: column;
      width: auto;
      height: auto;
      margin: 0px 41px;
    }

    &--button {
      margin-left: 41px;
      margin-bottom: 64px;
      margin: 20px 0 64px 41px;
      text-decoration: none;
    }
  }
}
