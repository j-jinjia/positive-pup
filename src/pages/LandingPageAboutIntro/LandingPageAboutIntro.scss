.landing-page__about-intro {

    &--greenbox{
        background-color: #F3FAF9;
        width: 1126px;
        height: 734px;
    }
}