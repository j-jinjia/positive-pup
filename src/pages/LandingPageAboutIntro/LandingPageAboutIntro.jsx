import "./LandingPageAboutIntro.scss";
import introImage from "../../assets/images/landing_page/landing_page_about_intro_img.png";
import Button from "../../components/Button/Button";
import paw from "../../assets/images/landing_page/paw_print.svg";
import { Link } from "react-router-dom";

const LandingPageAboutIntro = () => {
  return (
    <>
      <div className="landing-page__about-intro">
        <div className="landing-page__about-intro--greenbox">
          <img
            src={paw}
            className="landing-page__about-intro--paw"
            alt="paw print"
          />

          <h2 className="landing-page__about-intro--header">
            Why The Positive Pup?
          </h2>
          <p className="landing-page__about-intro--paragraph">
            {" "}
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mi
            malesuada tortor, tristique aenean ultricies. Integer aliquam mattis
            turpis justo, faucibus bibendum sagittis lacus ornare. Porttitor
            eget sed amet malesuada est dignissim magna. Amet vel tempor egestas
            at in mi, facilisi proin.
          </p>
          <Link to="/about" className="landing-page__about-intro--button">
            <Button isSecondary={true} buttonText="More About Me" />
          </Link>

          <img
            src={introImage}
            className="landing-page__about-intro--image"
            alt="Positive Pup owner, Melanie, with a dog"
          />
        </div>
      </div>
    </>
  );
};

export default LandingPageAboutIntro;
