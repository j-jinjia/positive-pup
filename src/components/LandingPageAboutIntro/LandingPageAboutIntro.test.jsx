import { customRender } from "../../utils/testUtils";
import LandingPageAboutIntro from "./LandingPageAboutIntro";

it("should render the LandingPageAboutIntro component", () => {
  const { container } = customRender(<LandingPageAboutIntro />);

  expect(container).toMatchSnapshot();
});
