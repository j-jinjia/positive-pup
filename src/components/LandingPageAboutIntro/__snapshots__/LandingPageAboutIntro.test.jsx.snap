// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the home page 1`] = `
<div>
  <div
    class="landing-page__about-intro"
  >
    <div
      class="landing-page__about-intro--greenbox"
    >
      <img
        alt="paw print"
        class="landing-page__about-intro--paw"
        src="paw_print.svg"
      />
      <h2
        class="landing-page__about-intro--header"
      >
        Why The Positive Pup?
      </h2>
      <p
        class="landing-page__about-intro--paragraph"
      >
         
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mi malesuada tortor, tristique aenean ultricies. Integer aliquam mattis turpis justo, faucibus bibendum sagittis lacus ornare. Porttitor eget sed amet malesuada est dignissim magna. Amet vel tempor egestas at in mi, facilisi proin.
      </p>
      <a
        class="landing-page__about-intro--button"
        href="/about"
      >
        <button
          class="button button--secondary"
        >
          More About Me
        </button>
      </a>
      <img
        alt="Positive Pup owner, Melanie, with a dog"
        class="landing-page__about-intro--image"
        src="landing_page_about_intro_img.png"
      />
    </div>
  </div>
</div>
`;
